### PR TITLE
cli: clean cmd migrated to docopts.

### PR DESCRIPTION
**Pull request by sergiusens**
This also involved moving `load_config` to the `yaml` module.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

### CODE REVIEW

**Comment by sergiusens** [comment]
This one has a lot of code movement

File: snapcraft/cmds.py [hunk 1]
```diff
@@ -61,7 +57,7 @@ def init(args):
 
 
 def shell(args):
-    config = _load_config()
+    config = snapcraft.yaml.load_config()
```
Comment (elopio) [inline snapcraft/cmds.py]: :) nice.

File: snapcraft/commands/clean.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Options:
+  -h --help             show this help message and exit.
+"""
+
+import os
+import logging
+import shutil
+import sys
+
+# Non-Standard Library modules
```
Comment (elopio) [inline snapcraft/commands/clean.py]: no need for the comment.

File: snapcraft/commands/clean.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+from docopt import docopt
+
+# Snapcraft modules
```
Comment (elopio) [inline snapcraft/commands/clean.py]: no need for the comment.

File: snapcraft/tests/test_commands_clean.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from unittest import mock
+
+# Non-Standard Library modules
```
Comment (elopio) [inline snapcraft/tests/test_commands_clean.py]: ctrl+k

File: snapcraft/tests/test_commands_clean.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+import fixtures
+
+# Snapcraft modules
```
Comment (elopio) [inline snapcraft/tests/test_commands_clean.py]: ctrl+k

File: snapcraft/yaml.py [hunk 6]
```diff
@@ -16,17 +16,23 @@
 
 import codecs
 import contextlib
-import jsonschema
 import logging
 import os
 import os.path
+import sys
+
+# Non-Standard Library modules
```
Comment (elopio) [inline snapcraft/yaml.py]: remove the comment, please.

File: snapcraft/yaml.py [hunk 7]
```diff
@@ -16,17 +16,23 @@
...
+import jsonschema
 import yaml
 
+# Snapcraft modules
```
Comment (elopio) [inline snapcraft/yaml.py]: same here.

**Comment by elopio** [comment]
Leaving my +1 here because I'm liking the reorg very much.
I just left some comments about the import comments, I find them noisy.

File: snapcraft/tests/test_commands_clean.py (hunk 4 above)
Comment (sergiusens) [inline snapcraft/tests/test_commands_clean.py]: No! `dd` :wink:

File: snapcraft/yaml.py [hunk 8]
```diff
@@ -345,3 +351,35 @@ def _expand_filesets_for(stage, properties):
...
+        _config = Config()
+        return _config
+    except SnapcraftYamlFileError as e:
+        logger.error(
+            'Could not find {}.  Are you sure you are in the right '
+            'directory?\nTo start a new project, use \'snapcraft '
+            'init\''.format(e.file))
+        sys.exit(1)
+    except SnapcraftSchemaError as e:
+        msg = 'Issues while validating snapcraft.yaml: {}'.format(e.message)
```
Comment (kyrofa) [inline snapcraft/yaml.py]: Just an inconsistency: `msg` isn't created in any of these other cases-- the string is passed directly to `logger.error()`. You can safely ignore this, I just wanted to point it out.

**Comment by kyrofa** [comment]
Looks good to me. I made one pedantic comment that can likely be ignored.

**Comment by sergiusens** [comment]
On Thu, Dec 3, 2015 at 10:31 AM, Kyle Fazzari <notifications@github.com>
wrote:

> Looks good to me. I made one pedantic comment that can likely be ignored.
>

I'll fix this when I rework the errors here.

**Pull request merged**